neurom/apps/cut_plane_detection.py

```diff
@@ -0,0 +1,191 @@
+# Copyright (c) 2015, Ecole Polytechnique Federale de Lausanne, Blue Brain Project
+# All rights reserved.
+#
+# This file is part of NeuroM <https://github.com/BlueBrain/NeuroM>
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+#     1. Redistributions of source code must retain the above copyright
+#        notice, this list of conditions and the following disclaimer.
+#     2. Redistributions in binary form must reproduce the above copyright
+#        notice, this list of conditions and the following disclaimer in the
+#        documentation and/or other materials provided with the distribution.
+#     3. Neither the name of the copyright holder nor the names of
+#        its contributors may be used to endorse or promote products
+#        derived from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+'''Module for the detection of the cut plane'''
+import logging
+import operator
+
+import numpy as np
+import neurom as nm
+
+from neurom import viewer
+from neurom.core import Tree
+from neurom.core.dataformat import COLS
+
+
+L = logging.getLogger(__name__)
+
+
+def _create_1d_distributions(points, bin_width):
+    '''Create point histograms along each axis
+
+    Parameters:
+        points: a np.ndarray of points
+        bin_width: the bin width
+
+    Returns: a dict of the X, Y and Z 1D histograms'''
+    min_, max_ = np.min(points, axis=0), np.max(points, axis=0)
+    hist = dict()
+    for i, plane in enumerate('XYZ'):
+        bins_min = np.arange(min_[i], max_[i] + bin_width, bin_width)
```

Comment:
bins_min has greater number than bins_max here no ?

Reply:
bins_min corresponds to the binning used for computing the first slice
bins_max correspond to the binning used for computing the last slice.
The thing is that, if the binning starting at first point is used for the first and last slice, it produces a bias. Indeed, even if the point distribution is symetrical, the first slice will have more counts than the last slice. So to have perfect symetrical conditions, the first slice count must be using a binning starting at the first point found, and the last slice must be using a binning starting at the last point found.
Not sure I am being clear...
Anyway, I'll rename to binning_first_slice, binning_last_slice